bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -56,9 +57,25 @@ void RequestProcessingState::handlePrimaryPreProcessed(const char *preProcessRes
       convertToArray(SHA3_256().digest(primaryPreProcessResult_, primaryPreProcessResultLen_).data());
 }
 
+void RequestProcessingState::detectNonDeterministicPreProcessing(const SHA3_256::Digest &newHash) const {
+  for (auto &hashArray : preProcessingResultHashes_)
+    if (newHash != hashArray.first) {
```

Comment:
Can we break the loop inside this `if` ?

Reply:
We want to log all the hash differences.

Reply:
Thanks for the clarification!